packages/xrpl/test/integration/transactions/signerListSet.test.ts

```diff
@@ -45,7 +45,10 @@ describe('SignerListSet', function () {
       }
       await testTransaction(testContext.client, tx, testContext.wallet)
 
-      const accountInfoResponse = await testContext.client.request({
+      const accountInfoResponse = await testContext.client.request<
+        AccountInfoRequest,
+        1
```

Comment:
Does this not default to 1?

Reply:
For TypeScript, it's not able to infer it's api_version 1. The type that gets assigned is the parent type `AccountInfoResponse` when type assertion isn't specified; it should be `AccountInfoV1Response` type.

Reply:
I put in a non-complete change to show:

https://github.com/XRPLF/xrpl.js/pull/2677/files#diff-a17f03a61f343bf490e5f4379ef5650ac8fcdc8c9dba35c8fd914c5502f83a7bR553-R563

- See if it's API version 2
  - If it is, go to V2 types
    - Check if account info
      - If it is, return account info v2
      - If it isn't, return the base types (which excludes account info)
  - If it isn't, go to V1 types
    - Check if account info
      - If it is, return account info v1
      - If it isn't, return the base types (which excludes account info)

Reply:
@justinr1234 Thanks for looking into it but unfortunately it didn't make a difference since type inference goes undetectable with TS Union and other complex types. And our codebase is filled with these types.